ukat/mapping/phase_contrast.py

```diff
@@ -0,0 +1,221 @@
+            self.resistive_index = ((mean_velocity_systole -
+                                     mean_velocity_diastole) /
+                                    mean_velocity_systole)
+            # Convert any nan values to 0
+            self.velocity_array = np.nan_to_num(self.velocity_array)
+            self.mask = np.nan_to_num(self.mask)
+        else:
+            raise ValueError('The input velocity_array should be 3D.')
+
+    def save_output_csv(self, path):
+        """
+        Save most of PhaseContrast class attributes into a csv file.
+
+        Parameters
+        ----------
+        path : str
+            Path to the desired csv file.
+        """
+        with open(path, 'w', newline='') as f:
+            w = csv.DictWriter(f, self.stats_table.keys())
+            w.writeheader()
+            for index in range(self.shape[-1]):
+                temp_dict = {}
+                for key in self.stats_table.keys():
+                    temp_dict[key] = self.stats_table[key][index]
+                w.writerow(temp_dict)
+        del temp_dict
+
+    def to_nifti(self, output_directory=os.getcwd(), base_file_name='Output',
+                 maps='all'):
+        """Exports the velocity array and the renal artery mask to NIFTI.
+
+        Parameters
+        ----------
+        output_directory : string, optional
+            Path to the folder where the NIFTI files will be saved.
+        base_file_name : string, optional
+            Filename of the resulting NIFTI. This code appends the extension.
+            Eg., base_file_name = 'Output' will result in 'Output.nii.gz'.
+        maps : list or 'all', optional
+            List of maps to save to NIFTI. This should either the string "all"
+            or a list of maps from ["phase_array", "mask", "velocity_array"].
+        """
+        os.makedirs(output_directory, exist_ok=True)
+        base_path = os.path.join(output_directory, base_file_name)
+        if maps == 'all' or maps == ['all']:
+            maps = ["velocity_array", "mask"]
+        if isinstance(maps, list):
+            for result in maps:
+                if result == 'velocity_array' or result == 'velocity array':
+                    velocity_nifti = nib.Nifti1Image(self.velocity_array,
+                                                     affine=self.affine)
+                    nib.save(velocity_nifti, base_path +
+                             '_velocity_array.nii.gz')
+                elif result == 'mask':
+                    mask_nifti = nib.Nifti1Image(self.mask.astype(int),
+                                                 affine=self.affine)
+                    nib.save(mask_nifti, base_path + '_mask.nii.gz')
+        else:
+            raise ValueError('No NIFTI file saved. The variable "maps" '
+                             'should be "all" or a list of maps from '
+                             '"["velocity_array", "mask"]".')
+
```

Comment:
I'd be tempted to add a method that generates the plots you've put in the notebook as they're a good sanity check.

Reply:
Agreed, will do that

Reply:
@alexdaniel654 in your opinion, should I write a method that creates a plot for each output or should one of the input arguments of this method be a stat. 

In other words, should we do `def plot_output(self):` or `def plot_output(self, stat):` where the first will display various plots and the 2nd will only display 1 plot associated to `stat.`

Reply:
How about having stat as an input but default to mean velocity and RBF as you had in the notebook i.e.
```python
def plot(self, stat='default'):
...
if stat == 'default':
    fig, (ax1, ax2) = plt.subplots(2, 1, figsize=(10, 10))
    ax1.plot(self.mean_velocity, 'ro-')
    ax1.set_ylabel('Velocity (cm/sec)')
    ax1.set_xlabel('Phase')
    ax1.set_title('Average Velocity Left Renal Artery')
    ax2.plot(self.rbf, 'b-')
    ax2.set_ylabel('RBF (ml/min)')
    ax2.set_xlabel('Phase')
    ax2.set_title('Average Renal Blood Flow Left Renal Artery')
elfi stat == 'mean_velocity`:
    fig, ax = plt.subplots(figsize=(10, 10))
    ax.plot(self.mean_velocity, 'ro-')
    ax.set_ylabel('Velocity (cm/sec)')
    ax.set_xlabel('Phase')
    ax.set_title('Average Velocity Left Renal Artery')
etc ...
```

Also, I'd probably just call it `.plot()`. Output is a given and just makes lines of code longer.

Pandas has a [plot method](https://github.com/pandas-dev/pandas/blob/v1.4.0/pandas/plotting/_core.py#L620-L1758) that might give you some inspiration.

Reply:
Brilliant, I like the idea, thanks for your feedback :thumbsup: